packages/auth/src/authClients/magicLink.ts

```diff
@@ -5,7 +5,7 @@ import type { AuthClient } from './'
 export type MagicLink = Magic
 export type MagicUser = MagicUserMetadata
 export interface AuthClientMagicLink extends AuthClient {
-  login(options: { email: string; showUI?: boolean }): Promise<any>
+  login(options: { email: string }): Promise<any>
```

Comment:
One thing @dthyresson said in the issue this PR addresses, #2361:

> That way, only if the developer doesn't want the confirm message, then they can explicitly set showUI: false

If we remove `showUI` entirely, the developer won't be able to opt out right?

Reply:
Yes, that's true.

it removes the `showUI` entirely.

That is why I made a PR on https://github.com/redwoodjs/playground-auth/pull/60 which would have worked with the current implementation of Magic in redwood.js.

It is about addressing the parameters, [loginWithMagicLink](https://docs.magic.link/client-sdk/web/api-reference#loginwithmagiclink) addresses `email`, `showUI`, but not the `redirectURI`. Similarly the [getIdToken](https://docs.magic.link/client-sdk/web/api-reference#getidtoken) doesn’t address the `lifespan` parameter.

https://github.com/redwoodjs/redwood/blob/main/packages/auth/src/authClients/magicLink.ts#L18

https://github.com/redwoodjs/redwood/blob/main/packages/auth/src/authClients/magicLink.ts#L27

Reply:
Sorry, I am confused. Shouldn't we default showUI to false?

If the point is to "reduce friction" I assume you do not want to have the popup message? Thus it's false by default and a developer can still opt-in to true to show message?

```js
export const magicLink = (client: MagicLink): AuthClientMagicLink => {
  return {
    type: 'magicLink',
    client,
    login: async ({ email, showUI = false }) =>
      await client.auth.loginWithMagicLink({ email, showUI }),
    logout: async () => {
      await client.user.logout()
    },
    signup: async ({ email, showUI = false }) =>
      await client.auth.loginWithMagicLink({ email, showUI }),
    getToken: async () => await client.user.getIdToken(),
    getUserMetadata: async () =>
      (await client.user.isLoggedIn()) ? await client.user.getMetadata() : null,
  }
}
```

Reply:
Hrmm, well I would have interpreted "reduce friction" here to mean something more like "batteries included" aka reduce developer time to increase adoption.   If showUI defaults to false, developers then need to bring their own UI to display to users so they know that something is happening. Whereas with showUI true, the magic sdk provides that for you with the iframe overlay shown above, which also hides itself once login is completed.

Reply:
Code has been updated per my review comments. See https://github.com/redwoodjs/redwood/pull/2629/files

Resolving this conversation